File: aspnetcore/reactApp/QueryFeld.ts
interface QueryFeldBase {
    feldKey: string,
    feldName: string,
}

export interface StringQueryFeld extends QueryFeldBase {
    kind: 'string',
}

export interface DatumQueryFeld extends QueryFeldBase {
    kind: 'datum',
}

export type QueryFeld = StringQueryFeld | DatumQueryFeld;

File: aspnetcore/Model/QueryFeld.cs
namespace aspnetcore.Model {
    public class QueryFeld {
        protected QueryFeld(string kind, string feldKey, string feldName) {
            Kind = kind;
            FeldKey = feldKey;
            FeldName = feldName;
        }
        public string Kind { get; }
        public string FeldKey { get; }
        public string FeldName { get; }
    }

    public class StringQueryFeld : QueryFeld {
        public StringQueryFeld(string feldKey, string feldName) : base("string", feldKey, feldName) {
        }
    }

    public class DatumQueryFeld : QueryFeld {
        public DatumQueryFeld(string feldKey, string feldName) : base("datum", feldKey, feldName) {
        }
    }
}
